### ISSUE TITLE
Kubernetes worker hangs at the end of flow run execution

### ISSUE THREAD

**Issue by desertaxle** [PrefectHQ/prefect-kubernetes#55]
# Expectation / Proposal

The Kubernetes worker hangs at the end of a flow run execution when tearing down the Event Replicator. Main thread gets blocked on [this line](https://github.com/PrefectHQ/prefect-kubernetes/blob/0835803fb099bfa185e195ee92b236beb1476f8a/prefect_kubernetes/events.py#L71) when joining the replicator thread to the main thread. Adding a `timeout` to the `.join()` call prevents the blocking, but the replicator thread still hangs around, and then the worker hangs on shutdown.

# Traceback / Example

- [ ] I would like to [help contribute](https://PrefectHQ.github.io/prefect-kubernetes/#contributing) a pull request to resolve this!

**Comment by madkinsz** [comment]
What is the thread doing? You should get a stack trace for all of the threads with `py-spy dump`

**Comment by bunchesofdonald** [comment]
The issue seems to be that the Kubernetes client only evaluates if it should stop after it receives an http response and since we want to exit _after_ the last response would be received the watch never stops, it just sits waiting on a response that will never come.

**Comment by EmilRex** [comment]
I _think_ I'm observing this too, although it could be a separate issue. When I run a flow, the worker hangs until the Finished Job TTL passes. For example, if I set the TTL to 300s and run a flow, the worker will function normally until the flow completes, and then it will become unhealthy until the job is cleaned up after 300s. If I lower the TTL to 60s, the same thing happens, except after 60s instead of 300s.
